src/streaming.ts

```diff
@@ -195,33 +198,51 @@ export class StreamProcessor {
   private async msgProcessor(
     msg: StreamMsg,
     fn: FetchFunction,
+    fetchType: string,
     setFn: (identifier: string, data: FeatureConfig | Segment) => void,
     delFn: (identifier: string) => void,
   ): Promise<void> {
-    this.log.info('Processing message', msg);
+    this.log.info(`Processing message with type ${fetchType}`, msg);
     try {
       if (msg.event === 'create' || msg.event === 'patch') {
-        const { data } = await fn(
-          msg.identifier,
-          this.environment,
-          this.cluster,
+        const { data } = await this.retryFetchOperation(
+          () => fn(msg.identifier, this.environment, this.cluster),
+          fetchType,
         );
         setFn(msg.identifier, data);
       } else if (msg.event === 'delete') {
         delFn(msg.identifier);
       }
     } catch (error) {
       this.log.error(
-        'Error while fetching data with identifier:',
+        `Error while fetching ${fetchType} with identifier:`,
         msg.identifier,
         error,
       );
-      throw error;
```

Comment:
We shouldn't have been doing this, as unless the Client was wrapped in the user's own try-catch logic, this would have crashed the customer's app